localstripe/server.py

```diff
@@ -191,7 +191,17 @@ async def auth_middleware(request, handler):
     return await handler(request)
 
 
-app = web.Application(middlewares=[error_middleware, auth_middleware])
+@web.middleware
+async def save_store_middleware(request, handler):
+    try:
+        return await handler(request)
+    finally:
+        if request.method in ('PUT', 'POST', 'DELETE'):
+            store.dump_to_disk()
+
+
+app = web.Application(middlewares=[error_middleware, auth_middleware,
+                                   save_store_middleware])
```

Comment:
Middlewares are applied in reverse order of this array (cf. [aiohttp doc](https://docs.aiohttp.org/en/stable/web_advanced.html#middlewares)).
It would be more logical to first call `auth_middleware` then `save_store_middleware` then `error_middleware` to me. What do you think?

Reply:
They are encapsulated in each other.
For what I understand, in the current code, an exception will not call `store.dump`, while in your proposition an exception will call `store.dump`.
I'm not sure of what I'm saving, but anyway it's the only concrete change I can see between both proposition.
Do you see anything else that would change?
You say it's more "logical", it's possible, but you don't give argument of "why". You have an idea in mind?

Reply:
I don't know about the order of calls (I haven't tried), and I don't have a real opinion about whether an exception should call `store.dump()` or not. Maybe it's better not to save to disk when an exception occurs, so that the saved state remains "clean"? I don't have a real opinion on this, so @bravier if you have something in mind, I'm curious too.

Reply:
What I had in mind when reading the doc was that the request handler was build by applying each item in this array of middlewares in reverse order.
It meant for me that, for a single request, we first want to authenticate this request then we try to save the store to disk if needed (once request is authenticated) and finally return an error in case of an issue. Which would give:
```py
app = web.Application(middlewares=[error_middleware, save_store_middleware,
                      auth_middleware])
```
But I didn't tested neither and it's just my understanding.
Since you tested it @H--o-l, I'm confident with your code :+1: